src/Internal/Service/PaymentProcessingService.php

```diff
@@ -134,6 +157,11 @@ public function get_authentication_redirect_url( $intent, int $order_id ) {
 	protected function create_payment_context( int $order_id, bool $manual_capture = false ): PaymentContext {
 		$context = new PaymentContext( $order_id );
 		$context->toggle_manual_capture( $manual_capture );
+		try {
+			$context->set_mode( $this->mode->is_test() ? 'test' : 'prod' );
+		} catch ( Exception $e ) {
+			$this->logger->error( 'Failed to fetch the mode: ' . $e );
```

Comment:
This exception is almost impossible to appear as late as the beginning of the payment process, as the mode will surely be initialized by this point. With that in mind, let's just `->set_mode( 'unknown' )` here, and add a comment that this is not an expected scenario.

Reply:
The `is_test` method throws `Exception` so I either have to deal with it at this point or keep bubbling it up, which looked really odd.

Reply:
I'm not saying to bubble the exception up:
```suggestion
			$context->set_mode( 'unknown' );
```

A simple setter to make sure something was set should be sufficient here.

Reply:
Done